### ISSUE TITLE
Exception using SDV with mlflow.pyfunc.load_model custom Python model

### ISSUE THREAD

**Issue by PJPRoche** [sdv-dev/SDV#1063]
### Environment Details

Please indicate the following details about the environment in which you found the bug:

* SDV version: 0.17.0
* Python version: 3.8
* Operating System: Linux 5.4.0-1090-azure / Windows 10

### Error Description

I am using MLFlow to train a SDV GAN neural network model and logging the experiment with all its artifacts, which includes the pickled custom GAN model class. I am using MLFlow in Azure Databricks for all the training.

After training, I am downloading all the artifacts so I can run some local inference. I am creating a conda environment using the inferred pip requirements (see the list in the above "System information" section) that gets bundled into the trained logged model artifacts. I then run the mlflow.pyfunc.load_model function and try get some predictions from the model.

I appreciate there are a number of factors here: platforms / environments / custom model classes which makes this hard to isolate and maybe for other to reproduce issue. However, I thought I would share in case others are having the same issue, or if someone has an idea how I can get to the bottom of the issue!! I have tried and debugged a lot of things but am not having any success in solving the problem.

Thanks for any help/suggestions

### Steps to reproduce

I have also raised this as an issue on the MLFlow GitHub page https://github.com/mlflow/mlflow/issues/6947 as it is not clear where the source of the issue lies.

This is the Databricks notebook I am using to train the model
```
# Databricks notebook source
pip install sdv==0.17.0

# COMMAND ----------

# DBTITLE 1,Install and Load Libraries
import logging
import mlflow
import mlflow.pyfunc

from mlflow.tracking.client import MlflowClient

from sdv.tabular import CTGAN
from sdv.demo import load_tabular_demo

from sys import version_info

# COMMAND ----------

# DBTITLE 1,Configure logging
# setting logging to DEBUG level whilst in development
logging.basicConfig(level=logging.DEBUG)
logger = logging.getLogger(__name__)

# turn off/set to error other modules
logger = logging.getLogger("py4j").setLevel(logging.ERROR)
logger = logging.getLogger("rdt").setLevel(logging.ERROR)
logger = logging.getLogger("urllib3v").setLevel(logging.ERROR)

# COMMAND ----------

mlflow_experiment_name = "/Shared/SyntheticGAN"

SDV_MODEL_NAME = "sdv_model.pkl"

# name of sub-folder to save the MLFlow model under the experiment id
MLFLOW_PYFUNC_MODEL_PATH = "sdv_mlflow_pyfunc"

# Parameter grid for Neural Network
PARAM_GRID = dict(
  epochs = 1
)

# COMMAND ----------

PYTHON_VERSION = "{major}.{minor}.{micro}".format(major=version_info.major,
                                                  minor=version_info.minor,
                                                  micro=version_info.micro)

mlflow.set_experiment(mlflow_experiment_name)

# COMMAND ----------

# Create an `artifacts` dictionary that assigns a unique name to the saved SDV model file.
# This dictionary will be passed to `mlflow.pyfunc.log_model`, which will copy the model file
# into the new MLflow experiment model's directory.
artifacts = {
    "sdv_model": SDV_MODEL_NAME
}

# COMMAND ----------

# Define the model class
# https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html#
# https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html#pyfunc-create-custom
class SDVWrapper(mlflow.pyfunc.PythonModel):
    """Defines the custom wrapper class."""
    def load_context(self, context):
        from sdv.tabular import CTGAN
        self.sdv_model = CTGAN.load(context.artifacts["sdv_model"])

    def predict(self, context, model_input):
        return self.sdv_model.sample(num_rows=model_input)

# COMMAND ----------

data = load_tabular_demo('student_placements')

# COMMAND ----------

# Start a new MLflow training run 
with mlflow.start_run() as run:
  
    run_id = "runs:/" + run.info.run_id + "/sdv_mlflow_pyfunc"
    
    model = CTGAN()
    model.fit(data)
    model.save(SDV_MODEL_NAME)
    
    # Log the MLflow Model
    # https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html#mlflow.pyfunc.log_model
    mlflow.pyfunc.log_model(
        artifact_path=MLFLOW_PYFUNC_MODEL_PATH, 
        python_model=SDVWrapper(), 
        artifacts=artifacts,
    )
  

# COMMAND ----------

run_id

# COMMAND ----------

model = mlflow.pyfunc.load_model(run_id)

# COMMAND ----------

model.predict(10)

```

This is the local python script I am using to make inferences with

```
import mlflow

logged_model = 'PATH_TO_LOCALLY_SAVED_TRAINED_MODEL_ARTIFACTS_FOLDER'

loaded_model = mlflow.pyfunc.load_model(logged_model)
print(loaded_model.predict(10))
```

### Stack trace

```
Traceback (most recent call last):
  File "C:\Users\######\Miniconda3\envs\sdv-demo\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\######\Miniconda3\envs\sdv-demo\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "c:\Users\######\.vscode\extensions\ms-python.python-2022.2.1924087327\pythonFiles\lib\python\debugpy\__main__.py", line 45, in <module>
    cli.main()
  File "c:\Users\######\.vscode\extensions\ms-python.python-2022.2.1924087327\pythonFiles\lib\python\debugpy/..\debugpy\server\cli.py", line 444, in main
    run()
  File "c:\Users\######\.vscode\extensions\ms-python.python-2022.2.1924087327\pythonFiles\lib\python\debugpy/..\debugpy\server\cli.py", line 285, in run_file
    runpy.run_path(target_as_str, run_name=compat.force_str("__main__"))
  File "C:\Users\######\Miniconda3\envs\sdv-demo\lib\runpy.py", line 265, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "C:\Users\######\Miniconda3\envs\sdv-demo\lib\runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "C:\Users\######\Miniconda3\envs\sdv-demo\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\######\repos\###########/run_models/sample_data_generator/run_mlflow_test.py", line 5, in <module>
    loaded_model = mlflow.pyfunc.load_model(logged_model)
  File "C:\Users\######\Miniconda3\envs\sdv-demo\lib\site-packages\mlflow\pyfunc\__init__.py", line 710, in load_model
    model_impl = importlib.import_module(conf[MAIN])._load_pyfunc(data_path)
  File "C:\Users\######\Miniconda3\envs\sdv-demo\lib\site-packages\mlflow\pyfunc\model.py", line 280, in _load_pyfunc
    python_model.load_context(context=context)
  File "<command-1315971304203110>", line 8, in load_context
  File "C:\Users\######\Miniconda3\envs\sdv-demo\lib\site-packages\sdv\tabular\base.py", line 891, in load
    model = cloudpickle.load(f)
TypeError: _restore() takes 3 positional arguments but 4 were given

**Comment by PJPRoche** [comment]
I am going to close this issue as I don't think it is related to SDV - it just so happened SDV was the library I was using when I encountered the issue. I worked with the Databricks team to try isolate it further but we were unable to get to the root cause. What did resolve the issue was migrating everything to Python 3.9 on Databricks 11.3 LTS. Maybe not the most satisfying solution but it works! I suspect the answer to why it was not working is related to the pickle protocol that cloudpickle uses - but I have not got the time to test this anymore ...
